source/slang/slang-check-stmt.cpp

```diff
@@ -247,16 +289,16 @@ namespace Slang
         stmt->condition = CheckExpr(stmt->condition);
         subContext.checkStmt(stmt->body);
 
-        // TODO(tfoley): need to check that all case tags are unique
-
-        // TODO(tfoley): check that there is at most one `default` clause
+        // check the case value exits within the switch
+        validateCaseStmts(stmt, getSink());
     }
 
     void SemanticsStmtVisitor::visitCaseStmt(CaseStmt* stmt)
     {
-        // TODO(tfoley): Need to coerce to type being switch on,
-        // and ensure that value is a compile-time constant
         auto expr = CheckExpr(stmt->expr);
+
+        // coerce to type being switch on, and ensure that value is a compile-time constant
+        auto exprVal = tryConstantFoldExpr(expr, ConstantFoldingKind::CompileTime, nullptr);
```

Comment:
tryConstantFoldExpr does not produce diagnostic if folding failed. We could try to test if exprVal is null and issue error if so. Hopefully this won't break the Falcor test this time.

Reply:
Well, unfortunately it did fail Falcor image test. So I pulled that check out of the PR :(